%  LaTeX support: latex@mdpi.com 
%  For support, please attach all files needed for compiling as well as the log file, and specify your operating system, LaTeX version, and LaTeX editor.

%=================================================================
\documentclass[journal,article,submit,pdftex,moreauthors]{Definitions/mdpi}
%\renewcommand{\linenumbers}{}
\usepackage{comment}
% For posting an early version of this manuscript as a preprint, you may use "preprints" as the journal and change "submit" to "accept". The document class line would be, e.g., \documentclass[preprints,article,accept,moreauthors,pdftex]{mdpi}. This is especially recommended for submission to arXiv, where line numbers should be removed before posting. For preprints.org, the editorial staff will make this change immediately prior to posting.

%--------------------
% Class Options:
%--------------------
%----------
% journal
%----------
% Choose between the following MDPI journals:
% acoustics, actuators, addictions, admsci, adolescents, aerospace, agriculture, agriengineering, agronomy, ai, algorithms, allergies, alloys, analytica, animals, antibiotics, antibodies, antioxidants, applbiosci, appliedchem, appliedmath, applmech, applmicrobiol, applnano, applsci, aquacj, architecture, arts, asc, asi, astronomy, atmosphere, atoms, audiolres, automation, axioms, bacteria, batteries, bdcc, behavsci, beverages, biochem, bioengineering, biologics, biology, biomass, biomechanics, biomed, biomedicines, biomedinformatics, biomimetics, biomolecules, biophysica, biosensors, biotech, birds, bloods, blsf, brainsci, breath, buildings, businesses, cancers, carbon, cardiogenetics, catalysts, cells, ceramics, challenges, chemengineering, chemistry, chemosensors, chemproc, children, chips, cimb, civileng, cleantechnol, climate, clinpract, clockssleep, cmd, coasts, coatings, colloids, colorants, commodities, compounds, computation, computers, condensedmatter, conservation, constrmater, cosmetics, covid, crops, cryptography, crystals, csmf, ctn, curroncol, currophthalmol, cyber, dairy, data, dentistry, dermato, dermatopathology, designs, diabetology, diagnostics, dietetics, digital, disabilities, diseases, diversity, dna, drones, dynamics, earth, ebj, ecologies, econometrics, economies, education, ejihpe, electricity, electrochem, electronicmat, electronics, encyclopedia, endocrines, energies, eng, engproc, ent, entomology, entropy, environments, environsciproc, epidemiologia, epigenomes, est, fermentation, fibers, fintech, fire, fishes, fluids, foods, forecasting, forensicsci, forests, foundations, fractalfract, fuels, futureinternet, futureparasites, futurepharmacol, futurephys, futuretransp, galaxies, games, gases, gastroent, gastrointestdisord, gels, genealogy, genes, geographies, geohazards, geomatics, geosciences, geotechnics, geriatrics, hazardousmatters, healthcare, hearts, hemato, heritage, highthroughput, histories, horticulturae, humanities, humans, hydrobiology, hydrogen, hydrology, hygiene, idr, ijerph, ijfs, ijgi, ijms, ijns, ijtm, ijtpp, immuno, informatics, information, infrastructures, inorganics, insects, instruments, inventions, iot, j, jal, jcdd, jcm, jcp, jcs, jdb, jeta, jfb, jfmk, jimaging, jintelligence, jlpea, jmmp, jmp, jmse, jne, jnt, jof, joitmc, jor, journalmedia, jox, jpm, jrfm, jsan, jtaer, jzbg, kidney, kidneydial, knowledge, land, languages, laws, life, liquids, literature, livers, logics, logistics, lubricants, lymphatics, machines, macromol, magnetism, magnetochemistry, make, marinedrugs, materials, materproc, mathematics, mca, measurements, medicina, medicines, medsci, membranes, merits, metabolites, metals, meteorology, methane, metrology, micro, microarrays, microbiolres, micromachines, microorganisms, microplastics, minerals, mining, modelling, molbank, molecules, mps, msf, mti, muscles, nanoenergyadv, nanomanufacturing, nanomaterials, ncrna, network, neuroglia, neurolint, neurosci, nitrogen, notspecified, nri, nursrep, nutraceuticals, nutrients, obesities, oceans, ohbm, onco, oncopathology, optics, oral, organics, organoids, osteology, oxygen, parasites, parasitologia, particles, pathogens, pathophysiology, pediatrrep, pharmaceuticals, pharmaceutics, pharmacoepidemiology, pharmacy, philosophies, photochem, photonics, phycology, physchem, physics, physiologia, plants, plasma, pollutants, polymers, polysaccharides, poultry, powders, preprints, proceedings, processes, prosthesis, proteomes, psf, psych, psychiatryint, psychoactives, publications, quantumrep, quaternary, qubs, radiation, reactions, recycling, regeneration, religions, remotesensing, reports, reprodmed, resources, rheumato, risks, robotics, ruminants, safety, sci, scipharm, seeds, sensors, separations, sexes, signals, sinusitis, skins, smartcities, sna, societies, socsci, software, soilsystems, solar, solids, sports, standards, stats, stresses, surfaces, surgeries, suschem, sustainability, symmetry, synbio, systems, taxonomy, technologies, telecom, test, textiles, thalassrep, thermo, tomography, tourismhosp, toxics, toxins, transplantology, transportation, traumacare, traumas, tropicalmed, universe, urbansci, uro, vaccines, vehicles, venereology, vetsci, vibration, viruses, vision, waste, water, wem, wevj, wind, women, world, youth, zoonoticdis 

%---------
% article
%---------
% The default type of manuscript is "article", but can be replaced by: 
% abstract, addendum, article, book, bookreview, briefreport, casereport, comment, commentary, communication, conferenceproceedings, correction, conferencereport, entry, expressionofconcern, extendedabstract, datadescriptor, editorial, essay, erratum, hypothesis, interestingimage, obituary, opinion, projectreport, reply, retraction, review, perspective, protocol, shortnote, studyprotocol, systematicreview, supfile, technicalnote, viewpoint, guidelines, registeredreport, tutorial
% supfile = supplementary materials

%----------
% submit
%----------
% The class option "submit" will be changed to "accept" by the Editorial Office when the paper is accepted. This will only make changes to the frontpage (e.g., the logo of the journal will get visible), the headings, and the copyright information. Also, line numbering will be removed. Journal info and pagination for accepted papers will also be assigned by the Editorial Office.

%------------------
% moreauthors
%------------------
% If there is only one author the class option oneauthor should be used. Otherwise use the class option moreauthors.

%---------
% pdftex
%---------
% The option pdftex is for use with pdfLaTeX. If eps figures are used, remove the option pdftex and use LaTeX and dvi2pdf.

%=================================================================
% MDPI internal commands
\firstpage{1} 
\makeatletter 
\setcounter{page}{\@firstpage} 
\makeatother
\pubvolume{1}
\issuenum{1}
\articlenumber{0}
\pubyear{2022}
\copyrightyear{2022}
%\externaleditor{Academic Editor: Firstname Lastname}
\datereceived{} 
%\daterevised{} % Only for the journal Acoustics
\dateaccepted{} 
\datepublished{} 
%\datecorrected{} % Corrected papers include a "Corrected: XXX" date in the original paper.
%\dateretracted{} % Corrected papers include a "Retracted: XXX" date in the original paper.
\hreflink{https://doi.org/} % If needed use \linebreak
%\doinum{}
%------------------------------------------------------------------
% The following line should be uncommented if the LaTeX file is uploaded to arXiv.org
%\pdfoutput=1

%=================================================================
% Add packages and commands here. The following packages are loaded in our class file: fontenc, inputenc, calc, indentfirst, fancyhdr, graphicx, epstopdf, lastpage, ifthen, lineno, float, amsmath, setspace, enumitem, mathpazo, booktabs, titlesec, etoolbox, tabto, xcolor, soul, multirow, microtype, tikz, totcount, changepage, attrib, upgreek, cleveref, amsthm, hyphenat, natbib, hyperref, footmisc, url, geometry, newfloat, caption

%=================================================================
%% Please use the following mathematics environments: Theorem, Lemma, Corollary, Proposition, Characterization, Property, Problem, Example, ExamplesandDefinitions, Hypothesis, Remark, Definition, Notation, Assumption
%% For proofs, please use the proof environment (the amsthm package is loaded by the MDPI class).

%=================================================================
% Full title of the paper (Capitalized)
\Title{A review on trending Machine Learning techniques for type 2 diabetes.}

% MDPI internal command: Title for citation in the left column
\TitleCitation{Title}

% Author Orchid ID: enter ID or remove command
\newcommand{\orcidauthorA}{0000-0000-0000-000X} % Add \orcidA{} behind the author's name
%\newcommand{\orcidauthorB}{0000-0000-0000-000X} % Add \orcidB{} behind the author's name

% Authors, for the paper (add full first names)
\Author{Firstname Lastname $^{1,\dagger,\ddagger}$\orcidA{}, Firstname Lastname $^{2,\ddagger}$ and Firstname Lastname $^{2,}$*}

%\longauthorlist{yes}

% MDPI internal command: Authors, for metadata in PDF
\AuthorNames{Firstname Lastname, Firstname Lastname and Firstname Lastname}

% MDPI internal command: Authors, for citation in the left column
\AuthorCitation{Lastname, F.; Lastname, F.; Lastname, F.}
% If this is a Chicago style journal: Lastname, Firstname, Firstname Lastname, and Firstname Lastname.

% Affiliations / Addresses (Add [1] after \address if there is only one affiliation.)
\address{%
$^{1}$ \quad Affiliation 1; e-mail@e-mail.com\\
$^{2}$ \quad Affiliation 2; e-mail@e-mail.com}

% Contact information of the corresponding author
\corres{Correspondence: e-mail@e-mail.com; Tel.: (optional; include country code; if there are multiple corresponding authors, add author initials) +xx-xxxx-xxx-xxxx (F.L.)}

% Current address and/or shared authorship
\firstnote{Current address: Affiliation 3.} 
\secondnote{These authors contributed equally to this work.}
% The commands \thirdnote{} till \eighthnote{} are available for further notes

%\simplesumm{} % Simple summary

%\conference{} % An extended version of a conference paper

% Abstract (Do not insert blank lines, i.e. \\) 
\abstract{A single paragraph of about 200 words maximum. For research articles, abstracts should give a pertinent overview of the work. We strongly encourage authors to use the following style of structured abstracts, but without headings: (1) Background: place the question addressed in a broad context and highlight the purpose of the study; (2) Methods: describe briefly the main methods or treatments applied; (3) Results: summarize the article's main findings; (4) Conclusions: indicate the main conclusions or interpretations. The abstract should be an objective representation of the article, it must not contain results which are not presented and substantiated in the main text and should not exaggerate the main conclusions.}

% Keywords
\keyword{keyword 1; keyword 2; keyword 3 (List three to ten pertinent keywords specific to the article; yet reasonably common within the subject discipline.)} 

% The fields PACS, MSC, and JEL may be left empty or commented out if not applicable
%\PACS{J0101}
%\MSC{}
%\JEL{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Diversity
%\LSID{\url{http://}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Applied Sciences
%\featuredapplication{Authors are encouraged to provide a concise description of the specific application or a potential application of the work. This section is not mandatory.}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Data
%\dataset{DOI number or link to the deposited data set if the data set is published separately. If the data set shall be published as a supplement to this paper, this field will be filled by the journal editors. In this case, please submit the data set as a supplement.}
%\datasetlicense{License under which the data set is made available (CC0, CC-BY, CC-BY-SA, CC-BY-NC, etc.)}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Toxins
%\keycontribution{The breakthroughs or highlights of the manuscript. Authors can write one or two sentences to describe the most important part of the paper.}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Encyclopedia
%\encyclopediadef{For entry manuscripts only: please provide a brief overview of the entry title instead of an abstract.}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{document}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\setcounter{section}{-1} %% Remove this when starting to work on the template.
\section{How to Use this Template}

The template details the sections that can be used in a manuscript. Note that the order and names of article sections may differ from the requirements of the journal (e.g., the positioning of the Materials and Methods section). Please check the instructions on the authors' page of the journal to verify the correct order and names. For any questions, please contact the editorial office of the journal or support@mdpi.com. For LaTeX-related questions please contact latex@mdpi.com.%\endnote{This is an endnote.} % To use endnotes, please un-comment \printendnotes below (before References). Only journal Laws uses \footnote.

% The order of the section titles is: Introduction, Materials and Methods, Results, Discussion, Conclusions for these journals: aerospace,algorithms,antibodies,antioxidants,atmosphere,axioms,biomedicines,carbon,crystals,designs,diagnostics,environments,fermentation,fluids,forests,fractalfract,informatics,information,inventions,jfmk,jrfm,lubricants,neonatalscreening,neuroglia,particles,pharmaceutics,polymers,processes,technologies,viruses,vision


\section{Introduction}

The introduction should  \cite{LAMA2021e07419}, \cite{KAVAKIOTIS2017104}, \cite{kopitar2020early}, \cite{Howlader}, \cite{ijerph18063317}, \cite{Lai}, \cite{fazakis}, \cite{Zou}, \cite{DeSilva}, \cite{Dinh}, \cite{zhang}, \cite{Fregoso}, \cite{Xiong}, \cite{Rufo}, \cite{Benita}, \cite{Dritsas}  briefly place the study in a broad context and highlight why it is important. It should define the purpose of the work and its significance. The current state of the research field should be reviewed carefully and key publications cited. Please highlight controversial and diverging hypotheses when necessary. Finally, briefly mention the main aim of the work and highlight the principal conclusions. As far as possible, please keep the introduction comprehensible to scientists outside your particular field of research. Citing a journal paper. Now citing a book reference  or other reference types . Please use the command  for the following MDPI journals, which use author--date citation: Administrative Sciences, Arts, Econometrics, Economies, Genealogy, Humanities, IJFS, Journal of Intelligence, Journalism and Media, JRFM, Languages, Laws, Religions, Risks, Social Sciences, Literature.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Diabetes}
Maybe some details about diabetes
\section{Machine Learning Background}
Maybe some details about Machine Learning Theory.
\section{Relevant Sections}
\subsection{Related Work}
Here we will review the two referenced review paper.
\subsection{Machine Learning applications in diabetes}
As mentioned before, the applications of Statistical Analysis and 
Machine Learning in healthcare and more specifically in diabetes 
condition  have demonstrated a steady rise in the last two decades, 
since the development of corresponding programming frameworks have 
enabled the easy storage, collection, processing, analysis of the 
massively available data quantity and employment of statistical and
Machine Learning models \cite{frank2005weka, scikit-learn,  
seabold2010statsmodels}. Regarding diabetes research field, the 
literature deals with the  identification of diabetic people, early or
long term (2-10 years) prognosis and diabetes complications
prediction or identification. Considering the prevention of diabetes, the ultimate
goal is the extraction of features (e.g markers) which are relevant to diabetes
occurrence. Then, in case that  these features are configurable, the patient could 
have available some suggestions to apply in his lifestyle or diet in order to 
minimize the risk of developing diabetes.   
 \par Our literature review is focused on
relatively new research articles or systematic reviews which are 
related with the context of our article e.g prediction of diabetes
mellitus or prediabetes utilizing demographic, anthropometric,
biometric, laboratory, nutritional, medical history, etc. data as
input features. The first mathematical approaches  over diabetes 
issue consisted of statistical risk scores exploiting questionnaires
filled by waves from the participants. Some of the famous ones risk scores
are Leicester Risk Assessment Score\cite{gray} developed by Leicester 
University and FINDRISC \cite{lindstrom} developed by University of 
Helsinki. The former utilizing a Logistic Regression model, take into account 
age, ethnicity, sex, first degree family history of diabetes, antihypertensive 
therapy or history of hypertension, waist circumference and BMI to predict 
current impaired glucose regulation or diabetes mellitus, achieving an AUC 
metric of 72\% and the latter -also exploited Logistic Regression-uses gender, 
age, BMI, use of blood pressure medication, history of high blood glucose, 
physical activity, daily  consumption of vegetables, fruits or berries and 
family history of diabetes to predict a 10-year development achieving an AUC 
metric of 86\%. We can observe at a first glance two variances of 
diabetes studies. The Leicster Risk aims to identify the current health condition, 
while FINDRISC tries to predict a long term prevalence.
There are also numerous researches that deal with deep learning
and  more specifically with image recognition for the classification of 
diabetic retinopathy, which is a typical complication and very well studied 
in the research field, using images from eye bulb as input 
\cite{Fregoso,KAVAKIOTIS2017104}. Another diabetes complications
studies utilizing Machine Learning and Deep Learning include neuropathy and nephropathy 
\cite{Fregoso,KAVAKIOTIS2017104}. Apart from classification 
problems there are also regression methods which are exploited 
for the prediction of Fasting Plasma Glycose or HbA1c levels, 
i.e. biomarkers that are the best indicators of abnormal 
glycose regulation and consequently diabetes  mellitus presence 
\cite{Fregoso,KAVAKIOTIS2017104,kopitar2020early}.
\par Delving more into literature that is more relevant with the purpose of this 
study we can observe an adequate quantity of high quality articles which will help
to understand a principal methodology in order to identify or predict diabetes 
development. Next, the chosen papers will be clustered based on their purpose, their key
methodologies will be in a more detailed context described and also each other compared
for advantages and disadvantages.
\par The current-state detection of diabetes, in the sense that the class variable and the
independent features values are registered the same time is studied in \cite{Howlader,
kopitar2020early, Lai, Zou, Dinh, zhang, Xiong, Rufo, Benita,  Dritsas}.
In \cite{Howlader} the dataset used is PIMA from UCI repository \cite{Dua:2019}, 
containing 768 records of healthy (500) and diabetic (268) Arizonan women over 21 
years old with target variable the diabetes presence. First, during the feature 
selection procedure, methods like information gain, gain ratio, gini index,ANOVA, 
$\chi^2$ test, an extension of Relief, correlation, fast correlation and filter 
subset evaluation where employed. Glucose levels, BMI, diabetes pedigree function 
and age was identified as the best features on average from the aforementioned 
techniques. Then, a variety of models was trained and tested on the different feature
subsets derived from the feature selection techniques using 10 fold cross validation.
The models probed were GAMBoost, regularized logistic regression, penalized 
multinomial regression, Bayesian generalized linear model, penalized logistic 
regression, generalized linear model, sparse distance weighted discrimination,
generalized boosted regression model and  Naive Bayes. The results showed that
there is not a particular model that yields the highest metrics (Accuracy, 
Kappa Statistics, AUC, Sensitivity, Specificity, Log loss) simultaneously.
Generalized additive model using LOESS yeld the best score in Friedman test,
achieving AUC 85.36\% and Sensitivity, Specificity 86\%, 60\% respectively. They
concluded that the aforementioned feature subset and Machine Learning model could
assist physicians and researchers to predict T2D, however this model should be 
assessed in bigger datasets  for detecting new potentially crucial features and 
compared with other high performance models. In \cite{Lai}, researchers trained a GBM
, a Random Forest and a Logistic Regression over a dataset containing 13,309 records
from healthy and patients. The input features were Age, Gender, FPG, BMI, Triglycerides,
Systolic pressure and LDL. First, the dataset was split in 80\% training set and
20\% testing set. Then, a misclassification cost matrix was constructed with a false
negative-false positive ratio equal to 3/1 and zero cost for correct predictions.
This cost matrix was used along with AUC as objective functions in order to tune the 
hyperapameters of the models using 10-fold cross validation. Due to the class imbalance
the cut-off point of decision boundary was adjusted such that the misclassification cost
is minimized. After this adjustment, each model with the tuned hyperapameters was 
trained on the entire training set and finally evaluated in the testing set. The
best model was GBM with tuned hyperapameters $\textrm{number of trees}=257$, 
$\textrm{interaction depth}=2$, $\textrm{min samples per leaf}= 75$, 
$\textrm{learning rate}=0.126$ and $\textrm{threshold=0.24}$ achieving AUC 84.7\%,
misclassification rate 18.9\%, Sensitivity 71.6\% and Specificity 83.7\%. Similarly,
the Logistic Regression achieved values of 84\%, 19.6\%, 73,4\% and 82.3\%,
respectively. For the Random Forest classifier the AUC value was 83.4\%. They 
summarized suggesting the incorporation of such models to online programms for 
further assistance of physicians during patient assessments.


\par As mentioned before, apart from classification problems, Machine Learning
can be applied to diabetes through Regression for estimation of predictive
biomarkers such as FPG (Fasting Plasma Glycose) and revelation of factors that
relate with the FPG variability. To this end, \cite{kopitar2020early} utilize
models of three conceptually different families such as boosting, bagging and
linear regression, because each family has a different capability to detect 
hidden patterns and important features. The dataset initially consisted of 
27,050 adults EHRs (Electronic Health Records) with no prior diabetes diagnosis
between 2014 and 2017. A first propose of the study is to compare the models
performance against FINDRISC, thus records that have missing values in any of
the features that included also in FINDRISC, were dropped. Assuming normal
distribution of the features, outlier detection took part using the formula
$\overline{X} \pm(3 \times \textrm{SD})$ and each outlier value was marked as
missing. Then, records and features tha had 50\% or greater percent of missing
values where dropped. Finally, the remaining missing values were imputed with
MICE method using Bayesian linear regression for numerical values, logistic   
regression for binary values and polytomous regression for variables with more
than two possible values. The preprocessing stage yields a final dataset of 3,723
records, 58 features and the FPG target variable. These features can be grouped
in the following four groups: lipid profile lab results (HDL, LDL, total 
cholesterol and triglycerides), social determinants of health (consumption of 
alcohol, smoking, dietary habits, stress), cardiovascular variables (blood 
pressure measurements, atrial fibrillation history) and history of other health
conditions (stroke, hypertension, colon cancer). As the final step before 
piping the data into the Machine Learning models, is the partition 
of the data into 6-months intervals (T6, T12, T18, T24, T30) 
according to the submission date of each record, thus 5 subdatasets where created
and each Machine Learning model was trained in each subdataset and validated using
100 times random sampling with replacement (bootstrap). In each run the remaining
unselected samples were used to test the model. Linear Regression performed with 
the lowest Root Mean Squared Error 0.838 (95\% CI 0.814-0.862) trained on only 7 
features which are common to the FINDRISC. Next, RF achieved a value of 0.842 
(95\% CI 0.818-0.866), LightGBM 0.846 (95\% CI 0.821-0.871), Glmnet 0.859 
(95\% CI 0.834-0.884) and XGBoost 0.881 (95\% 0.856-0.907). When the whole dataset
where available for training and testing, RF performed the lowest RMSE at 0.745 
(95\% CI 0.733-0.757) followed by Glmnet at 0.747(95\% CI 0.734-0.759), while 
XGBoost performed the worst value at 0.760 (95\% CI 0.748-0.772). The regression 
capability of each model was measured through $R^{2}$ coefficient, to measure how
well the regressor fits the actual FPG value given the input features. For only 6 
month data available linear model performed the best with an average value of 
0.310, while RF performed the best for 18 and 30 months data available achieving a 
mean value of 0.340 and 0.368 respectively. In contrary, XGBoost performed the 
worst in all three time points (6-18-30 months of available data) achieving a mean
value of 0.241, 0.293 and 0.343 respectively, despite its general superiority. 
Apart from the regression capability, the authors tested the classification 
capability using the cut-off FPG value of 6.1 mmol/L. The best AUC value demonstrated
by Glmnet at 0.836 on average  through the five time points, while the worst 
performance model was again XGBoost yielding a value of 0.8142. In terms of AUPRC,
linear regression performed best with an average value of 0.6948 through the five
time points, while was the worst model with a value of 0.6576. Finally, the feature 
importance was assessed for every model through the five time point datasets
using different metrics (because each model has different structure) like 
$\beta \textrm{-coefficient}$, permutation importance on MSE or on Accuracy and 
variance gain. Triglycerides levels was assessed as the most important feature on
LightGBM, while the remaining  three models have Hyperglycemia history. For the
next lower-importance features, even if there are some differences in the ranking,
Age, HDL cholisterol, LDL cholisterol, Total cholisterol, Systolic pressure, Diastolic
pressure and weight are the in the top 10.  They concluded that, the more data 
available, the better stability do models have, even if from this research none
new FPG related feature revealed, apart from those already clinical derived. LightGBM
performed the most stable results through the multiple evaluations.
\par The long term diabetes prediction in the sense  that the class variable is filled
many years after the features are filled, utilizing the baseline-followup method, is 
studied in \cite{lindstrom, Dritsas}. In \cite{lindstrom} the study examined a cohort
of 7949 people with known and unknown family history of diabetes. The features involved
were questionnaires about lifestyle, socioeconomic and psychosocial matters, along
with measurements of plasma glucose and insulin in an oral glucose
tolerance test (OGTT), glycosylated haemoglobin (HbA1c), blood pressure,
weight, height and hip circumference. In the baseline study, T2D
was diagnosed in 51 women and 66 men, and prediabetes in 219 women
and 259 men. A $1^{\textrm{st}}$ follow-up study was carried out 8–10 years later,
and a $2^{\textrm{nd}}$ follow-up about 20 years later, with at least 70\% participation.
At the $1^{\textrm{st}}$ follow-up, they found 102 women and 171 men with T2D, and
399 women and 522 men with prediabetes, and at the $2^{\textrm{nd}}$ follow-up 230
women and 326 men with T2D and 615 women and 522 men with
prediabetes. Those with diagnosed T2D at the baseline and the $1^{\textrm{st}}$ followup
were not called to follow-up later. The dataset was partitioned
in 3 sets a training set, a validation set and a test set. The classifier utilized was Random 
Forest to predict the individual diabetes type 2 development after 10 years of the measurement.
SHAP TreeExplainer was used to build an interpretable Machine Learning model, in order to find 
factors that relate with high or low diabetes risk. Hyperparameter tuning using 5 fold cross 
validation into the validation set took part in order to find the best hyperparameters set for 
the Random Forest, having as objective function a combination of AUC and robustness. This
function is defined as: 
\begin{equation}
	S_l=AUC_{l}^{val}\cdot(1-\Theta_{l})
\end{equation}
Where
\begin{equation}
	\Theta_{l}=\mu(\sigma(X_{ijkl}^{'}))
\end{equation} 
and 
\begin{equation}
	X_{ijkl}^{'}=\frac{X_{ijkl}-\mu(X_{ijkl})}{\sigma(X_{ijkl})}
\end{equation}.
$X_{ijkl}$ is a tensor of SHAP values per person i, feature j, cross
validation split k and parameter set l. Then $X_{ijkl}^{'}$ is the standardized
tensor with zero mean and unit variance. Finally, $\Theta_{l}$ is the mean
variance of o SHAP values per hyperapameters set l. The best hyperparameters
set was: $\textrm{number of estimators} = 120, \textrm{min samples leaf}=125, \textrm{max depth}=4 \textrm{ and
number of models} = 30 $, achieving a robustness value of $ S=0.630 $ and
a value of AUC at $0.779$. According to the SHAP values analysis, the features that increase
the risk are: family history of diabetes, high waist-hip  ratio, high BMI, increased Systolic
pressure, increased Diastolic pressure, low physical activity and male gender. On the other
hand the features that decrease the risk are: exercise, higher socioeconomic strata and lower
age. Also with the help of a SHAP force plot, personalized risk profiles where extracted in order
to assess the individual risk score, which is called \textit{output value} and  revealed 
the features that have the largest impact in the individual risk score. Finally, they suggest 
this method to be probed in the primary health care in order to improve diabetes care.
In \cite{ijerph18063317} a dataset of 500,000 records from Hanaro medical foundation containing 
diagnostic results and questionnairies, was utilized to conduct a multi classification 
experiment for prediction of prediabetic,  diabetic and normal people in the following 1 year. 
People included in the dataset are those who had at least two years of continuous annual medical
check-ups during the follow-up period, however those who had been diagnosed with diabetes, 
hyperlipidimia, hypertension or took medication for those diseases were excluded. During the 
preprocessing step, exclusion of records with at least one null value, employment of majority 
undersampling and Synthetic Minority Oversampling techniques, ANOVA and  chi-square test, as well as
Recursive Feature Elimination using an impurity metric to rank feature importance, were conducted.
The selected features were fasting plasma glucose (FPG), body mass index (BMI), Gamma
glutamyl transpeptidase (gamma-GTP), triglycerides, sex, age, uric acid, hemoglobin A1c
(HbA1c), smoking, drinking, physical activity, and family history. Smoking status was
divided into “currently smoking regularly”, “never smoked” and “had quit smoking”.
Physical activity indicates the number of days the subject has engaged in physical exercise
such as running, hillwalking, climbing stairs, jump roping for a minimum of 20 min. Family
history with diabetes considers only parents and siblings diagnosed with T2D and drinking
indicates the number of days the subject consumed alcoholic drinks. Then, RF, XGBoost and SVM were
trained and tested on the dataset. First, two 10-fold cross validation grid searches for 
hyperaparemeter tunning were employed. The first in a more general grid, but the second focused 
on the best - more specific neighborhoods of good parameters values derived from the first.
In addition, a Logistic Regression, a Stacking classifier, a Soft Voting
classifier and a confusion matrix based classifier were employed. For the testing
phase, 10-fold cross validation repeated 10 times and the average value for each
metric such as accuracy, precision, recall, F1-score, MCC and KC was calculated.
The results shown none significant difference between the models performance. All
metrics were in the range 0.71-0.75. Then RF, XGBoost and SVM, were trained
using data from two years, three years and four years before, which where 
added each one to the previous set. The accuracy performance showed significant 
improvement for every model. As a final test, the 12-features set was 
compared to a set of very classic diabetes predictors such as FPG, HbA1c, BMI, age, and
sex into the different four datasets, which where previously introduced .
The 10-repetion evaluation of accuracy through the different timepoint datasets
showed significant superiority of the 12-features subset over the 5-features (
e.g. using data from the last four years the accuracy values were 0.81 for the
former feature set and 0.77 for the latter set). The authors concluded that clinicians
must pay attention to the changes in gamma-GTP, uric acid, and triglycerides over the
years, as well as these models can work as decision support systems for practioners 
and diabetes educators.
In \cite{fazakis}, researchers evaluated a variety of single and ensemble models
on ELSA dataset to predict type 2 diabetes occurrence. The dataset contains a variety
of biometric, anthropometric, hematological, lifestyle, sociodemographic and performance 
index variables. A number of different feature selection techniques were employed
such as LASSO, correlation and Greedy stepwise. The selected method was Greedy
stepwise with Naive Bayes and after the addition of some extra features the final
dataset consisted of 34 input features and the class variable indicating the diabetes
condition of  the person. Random undersampling was conducted in order to have diabetic
distribution per age group similar to real life. This yeld a final dataset of 2,331
records. For the evaluation of the models the procedure consists of creating 
10 datasets from the existing using stratified train/test split with proportion 
70/30 respecitively. Logistic Regression, Naive Bayes, Decision Tree, Random Forest,
Artificial Neural Network, Deep Neural Network and three ensembles of Random Forest
and Logistic Regression, namely Stacking, Voting and Weighted Voting, were employed.
Due to class imbalance the method of adjusting threshold were conducted for each
model having as objective function the $J$, Younden Index, which is the sum of
specificity and sensitivity. For the Weighted Soft Voting, a biobjective
optimization problem was solved in order to calculate  the best weights for 
RF and LR, which maximize the sensitivity and AUC. Indeed the Weighted classifier
produced the best results in terms of AUC with a value of 0.884. In addition,
the Sensitivity, Specificity +PV, -PV, +LR, -LR, were 0.856, 0.798, 0.449, 0.967 and
4.245 respectively. Finally, the concluded that due to the superiority of ensemble 
models can be embedded into recommendation systems to prevent patients from development
of diabetes. In \cite{Zou} a physical, 138,000-records dataset containing 
14 features such as age, pulse rate, breathe, left systolic pressure, right systolic 
pressure, left diastolic pressure, right diastolic pressure, height, weight, physique
index, fasting glucose, waistline, LDL and HDL was used. Five subdatasets were created with 
random sampling in order to train the models five times and then calculate the average 
performance into a independent testing set using 5 fold cross validation. The models
which were trained are Random Forest, Decision Tree and a Neural Network. The models
were evaluated in different subsets of features using feature selection techniques like
PCA, mRMR, without fasting glycose and only fasting glycose. Using all features every model
achieved the best accuracy, while excluding fasting glycose trained the worst models. In the 
first case the accuracy of RF, Decision Tree and Neural network was 0.8084, 0.7853 and 
0.7841 respectively. Using only fasting glycose as input feature trains the models still better
than PCA and mRMR techniques, yielding accuracy at 0.7597, 0.761 and 0.7572 for each model
respectively. They concluded that fasting blood glycose is a very good predictor of diabetes, however
adding more features gives better performing models. As feature work they propose 
the extraction of indicators importance and the classification of the speicific diabetes 
type.  



%\section{Materials and Methods}
Materials and Methods should be described with sufficient details to allow others to replicate and build on published results. Please note that publication of your manuscript implicates that you must make all materials, data, computer code, and protocols associated with the publication available to readers. Please disclose at the submission stage any restrictions on the availability of materials or information. New methods and protocols should be described in detail while well-established methods can be briefly described and appropriately cited.

Research manuscripts reporting large datasets that are deposited in a publicly avail-able database should specify where the data have been deposited and provide the relevant accession numbers. If the accession numbers have not yet been obtained at the time of submission, please state that they will be provided during review. They must be provided prior to publication.

Interventionary studies involving animals or humans, and other studies require ethical approval must list the authority that provided approval and the corresponding ethical approval code.
\begin{quote}
This is an example of a quote.
\end{quote}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\section{Results}

This section may be divided by subheadings. It should provide a concise and precise description of the experimental results, their interpretation as well as the experimental conclusions that can be drawn.
\subsection{Subsection}
\subsubsection{Subsubsection}

Bulleted lists look like this:
\begin{itemize}
\item	First bullet;
\item	Second bullet;
\item	Third bullet.
\end{itemize}

Numbered lists can be added as follows:
\begin{enumerate}
\item	First item; 
\item	Second item;
\item	Third item.
\end{enumerate}

The text continues here. 

\subsection{Figures, Tables and Schemes}

All figures and tables should be cited in the main text as Figure~\ref{fig1}, Table~\ref{tab1}, Table~\ref{tab2}, etc.

\begin{figure}[H]
\includegraphics[width=10.5 cm]{Definitions/logo-mdpi}
\caption{This is a figure. Schemes follow the same formatting. If there are multiple panels, they should be listed as: (\textbf{a}) Description of what is contained in the first panel. (\textbf{b}) Description of what is contained in the second panel. Figures should be placed in the main text near to the first time they are cited. A caption on a single line should be centered.\label{fig1}}
\end{figure}   
\unskip

\begin{table}[H] 
\caption{This is a table caption. Tables should be placed in the main text near to the first time they are~cited.\label{tab1}}
\newcolumntype{C}{>{\centering\arraybackslash}X}
\begin{tabularx}{\textwidth}{CCC}
\toprule
\textbf{Title 1}	& \textbf{Title 2}	& \textbf{Title 3}\\
\midrule
Entry 1		& Data			& Data\\
Entry 2		& Data			& Data\\
\bottomrule
\end{tabularx}
\end{table}
\unskip

\begin{table}[H]
\caption{This is a wide table.\label{tab2}}
	\begin{adjustwidth}{-\extralength}{0cm}
		\newcolumntype{C}{>{\centering\arraybackslash}X}
		\begin{tabularx}{\fulllength}{CCCC}
			\toprule
			\textbf{Title 1}	& \textbf{Title 2}	& \textbf{Title 3}     & \textbf{Title 4}\\
			\midrule
			Entry 1		& Data			& Data			& Data\\
			Entry 2		& Data			& Data			& Data \textsuperscript{1}\\
			\bottomrule
		\end{tabularx}
	\end{adjustwidth}
	\noindent{\footnotesize{\textsuperscript{1} This is a table footnote.}}
\end{table}

%\begin{listing}[H]
%\caption{Title of the listing}
%\rule{\columnwidth}{1pt}
%\raggedright Text of the listing. In font size footnotesize, small, or normalsize. Preferred format: left aligned and single spaced. Preferred border format: top border line and bottom border line.
%\rule{\columnwidth}{1pt}
%\end{listing}

Text.

Text.

\subsection{Formatting of Mathematical Components}

This is the example 1 of equation:
\begin{linenomath}
\begin{equation}
a = 1,
\end{equation}
\end{linenomath}
the text following an equation need not be a new paragraph. Please punctuate equations as regular text.
%% If the documentclass option "submit" is chosen, please insert a blank line before and after any math environment (equation and eqnarray environments). This ensures correct linenumbering. The blank line should be removed when the documentclass option is changed to "accept" because the text following an equation should not be a new paragraph.

This is the example 2 of equation:
\begin{adjustwidth}{-\extralength}{0cm}
\begin{equation}
a = b + c + d + e + f + g + h + i + j + k + l + m + n + o + p + q + r + s + t + u + v + w + x + y + z
\end{equation}
\end{adjustwidth}

% Example of a page in landscape format (with table and table footnote).
%\startlandscape
%\begin{table}[H] %% Table in wide page
%\caption{This is a very wide table.\label{tab3}}
%	\begin{tabularx}{\textwidth}{CCCC}
%		\toprule
%		\textbf{Title 1}	& \textbf{Title 2}	& \textbf{Title 3}	& \textbf{Title 4}\\
%		\midrule
%		Entry 1		& Data			& Data			& This cell has some longer content that runs over two lines.\\
%		Entry 2		& Data			& Data			& Data\textsuperscript{1}\\
%		\bottomrule
%	\end{tabularx}
%	\begin{adjustwidth}{+\extralength}{0cm}
%		\noindent\footnotesize{\textsuperscript{1} This is a table footnote.}
%	\end{adjustwidth}
%\end{table}
%\finishlandscape

% Example of a figure that spans the whole page width. The same concept works for tables, too.
\begin{figure}[H]
\begin{adjustwidth}{-\extralength}{0cm}
\centering
\includegraphics[width=13.5cm]{Definitions/logo-mdpi}
\end{adjustwidth}
\caption{This is a wide figure.\label{fig2}}
\end{figure}  

Please punctuate equations as regular text. Theorem-type environments (including propositions, lemmas, corollaries etc.) can be formatted as follows:
%% Example of a theorem:
\begin{Theorem}
Example text of a theorem.
\end{Theorem}

The text continues here. Proofs must be formatted as follows:

%% Example of a proof:
\begin{proof}[Proof of Theorem 1]
Text of the proof. Note that the phrase ``of Theorem 1'' is optional if it is clear which theorem is being referred to.
\end{proof}
The text continues here.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Discussion}

Authors should discuss the results and how they can be interpreted from the perspective of previous studies and of the working hypotheses. The findings and their implications should be discussed in the broadest context possible. Future research directions may also be highlighted.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Conclusions}

This section is not mandatory, but can be added to the manuscript if the discussion is unusually long or complex.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Future Directions}

This section is not mandatory, but may be added if there are patents resulting from the work reported in this manuscript.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\vspace{6pt} 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% optional
%\supplementary{The following supporting information can be downloaded at:  \linksupplementary{s1}, Figure S1: title; Table S1: title; Video S1: title.}

% Only for the journal Methods and Protocols:
% If you wish to submit a video article, please do so with any other supplementary material.
% \supplementary{The following supporting information can be downloaded at: \linksupplementary{s1}, Figure S1: title; Table S1: title; Video S1: title. A supporting video article is available at doi: link.}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\authorcontributions{For research articles with several authors, a short paragraph specifying their individual contributions must be provided. The following statements should be used ``Conceptualization, X.X. and Y.Y.; methodology, X.X.; software, X.X.; validation, X.X., Y.Y. and Z.Z.; formal analysis, X.X.; investigation, X.X.; resources, X.X.; data curation, X.X.; writing---original draft preparation, X.X.; writing---review and editing, X.X.; visualization, X.X.; supervision, X.X.; project administration, X.X.; funding acquisition, Y.Y. All authors have read and agreed to the published version of the manuscript.'', please turn to the  \href{http://img.mdpi.org/data/contributor-role-instruction.pdf}{CRediT taxonomy} for the term explanation. Authorship must be limited to those who have contributed substantially to the work~reported.}

\funding{Please add: ``This research received no external funding'' or ``This research was funded by NAME OF FUNDER grant number XXX.'' and  and ``The APC was funded by XXX''. Check carefully that the details given are accurate and use the standard spelling of funding agency names at \url{https://search.crossref.org/funding}, any errors may affect your future funding.}

\institutionalreview{In this section, you should add the Institutional Review Board Statement and approval number, if relevant to your study. You might choose to exclude this statement if the study did not require ethical approval. Please note that the Editorial Office might ask you for further information. Please add “The study was conducted in accordance with the Declaration of Helsinki, and approved by the Institutional Review Board (or Ethics Committee) of NAME OF INSTITUTE (protocol code XXX and date of approval).” for studies involving humans. OR “The animal study protocol was approved by the Institutional Review Board (or Ethics Committee) of NAME OF INSTITUTE (protocol code XXX and date of approval).” for studies involving animals. OR “Ethical review and approval were waived for this study due to REASON (please provide a detailed justification).” OR “Not applicable” for studies not involving humans or animals.}

\informedconsent{Any research article describing a study involving humans should contain this statement. Please add ``Informed consent was obtained from all subjects involved in the study.'' OR ``Patient consent was waived due to REASON (please provide a detailed justification).'' OR ``Not applicable'' for studies not involving humans. You might also choose to exclude this statement if the study did not involve humans.

Written informed consent for publication must be obtained from participating patients who can be identified (including by the patients themselves). Please state ``Written informed consent has been obtained from the patient(s) to publish this paper'' if applicable.}

\dataavailability{In this section, please provide details regarding where data supporting reported results can be found, including links to publicly archived datasets analyzed or generated during the study. Please refer to suggested Data Availability Statements in section ``MDPI Research Data Policies'' at \url{https://www.mdpi.com/ethics}. If the study did not report any data, you might add ``Not applicable'' here.} 

\acknowledgments{In this section you can acknowledge any support given which is not covered by the author contribution or funding sections. This may include administrative and technical support, or donations in kind (e.g., materials used for experiments).}

\conflictsofinterest{Declare conflicts of interest or state ``The authors declare no conflict of interest.'' Authors must identify and declare any personal circumstances or interest that may be perceived as inappropriately influencing the representation or interpretation of reported research results. Any role of the funders in the design of the study; in the collection, analyses or interpretation of data; in the writing of the manuscript; or in the decision to publish the results must be declared in this section. If there is no role, please state ``The funders had no role in the design of the study; in the collection, analyses, or interpretation of data; in the writing of the manuscript; or in the decision to publish the~results''.} 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Optional
\sampleavailability{Samples of the compounds ... are available from the authors.}

%% Only for journal Encyclopedia
%\entrylink{The Link to this entry published on the encyclopedia platform.}

\abbreviations{Abbreviations}{
The following abbreviations are used in this manuscript:\\

\noindent 
\begin{tabular}{@{}ll}
MDPI & Multidisciplinary Digital Publishing Institute\\
DOAJ & Directory of open access journals\\
TLA & Three letter acronym\\
LD & Linear dichroism
\end{tabular}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Optional
\appendixtitles{no} % Leave argument "no" if all appendix headings stay EMPTY (then no dot is printed after "Appendix A"). If the appendix sections contain a heading then change the argument to "yes".
\appendixstart
\appendix
\section[\appendixname~\thesection]{}
\subsection[\appendixname~\thesubsection]{}
The appendix is an optional section that can contain details and data supplemental to the main text---for example, explanations of experimental details that would disrupt the flow of the main text but nonetheless remain crucial to understanding and reproducing the research shown; figures of replicates for experiments of which representative data are shown in the main text can be added here if brief, or as Supplementary Data. Mathematical proofs of results not central to the paper can be added as an appendix.

\begin{table}[H] 
\caption{This is a table caption.\label{tab5}}
\newcolumntype{C}{>{\centering\arraybackslash}X}
\begin{tabularx}{\textwidth}{CCC}
\toprule
\textbf{Title 1}	& \textbf{Title 2}	& \textbf{Title 3}\\
\midrule
Entry 1		& Data			& Data\\
Entry 2		& Data			& Data\\
\bottomrule
\end{tabularx}
\end{table}

\section[\appendixname~\thesection]{}
All appendix sections must be cited in the main text. In the appendices, Figures, Tables, etc. should be labeled, starting with ``A''---e.g., Figure A1, Figure A2, etc.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{adjustwidth}{-\extralength}{0cm}
%\printendnotes[custom] % Un-comment to print a list of endnotes

\reftitle{References}

% Please provide either the correct journal abbreviation (e.g. according to the “List of Title Word Abbreviations” http://www.issn.org/services/online-services/access-to-the-ltwa/) or the full name of the journal.
% Citations and References in Supplementary files are permitted provided that they also appear in the reference list here. 

%=====================================
% References, variant A: external bibliography
%=====================================
\bibliography{references.bib}

%=====================================
% References, variant B: internal bibliography
%=====================================

% If authors have biography, please use the format below
%\section*{Short Biography of Authors}
%\bio
%{\raisebox{-0.35cm}{\includegraphics[width=3.5cm,height=5.3cm,clip,keepaspectratio]{Definitions/author1.pdf}}}
%{\textbf{Firstname Lastname} Biography of first author}
%
%\bio
%{\raisebox{-0.35cm}{\includegraphics[width=3.5cm,height=5.3cm,clip,keepaspectratio]{Definitions/author2.jpg}}}
%{\textbf{Firstname Lastname} Biography of second author}

% For the MDPI journals use author-date citation, please follow the formatting guidelines on http://www.mdpi.com/authors/references
% To cite two works by the same author: \citeauthor{ref-journal-1a} (\citeyear{ref-journal-1a}, \citeyear{ref-journal-1b}). This produces: Whittaker (1967, 1975)
% To cite two works by the same author with specific pages: \citeauthor{ref-journal-3a} (\citeyear{ref-journal-3a}, p. 328; \citeyear{ref-journal-3b}, p.475). This produces: Wong (1999, p. 328; 2000, p. 475)

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% for journal Sci
%\reviewreports{\\
%Reviewer 1 comments and authors’ response\\
%Reviewer 2 comments and authors’ response\\
%Reviewer 3 comments and authors’ response
%}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\end{adjustwidth}
\end{document}

